presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -83,7 +101,22 @@ public void onSuccess(BaseResponse<TaskInfo> result)
                             @Override
                             public void onFailure(Throwable t)
```

Comment:
When is onFailure triggered?

I don't think we ever throw an exception when there is any valid json response from the cpp server. Maybe we should trigger `onFailure` when there is also a 5XX response from cpp.

Reply:
@miaoever Do you have an answer to @vermapratyush 's question?

Reply:
@vermapratyush @vermapratyush  That's a good point. Just changed the taskInfoFetcher to throw exception when `familyForStatusCode(response.getStatusCode()) != HttpStatus.Family.SUCCESSFUL`[PrestoSparkHttpTaskClient.java](https://github.com/prestodb/presto/pull/19580/files#diff-ffd5ff45029ffab7e81016f03c7eeecb6a47c95b86ea5f7bd50df50fb351de7a) .

Also updated the unit test to demonstrate it.